### DISCUSSION TITLE
Object Browser filters not working

### DISCUSSION THREAD

**Discussion by dougnlamb** [codefori/vscode-ibmi#623] [Q&A] [answered]
I have tried creating several object browser filters, but no objects are displayed.  I have mirrored my filters from RDi as follows:

**RDi Example**
![image](https://user-images.githubusercontent.com/5246077/162847452-76b43ec1-0e66-4955-a972-d764b0a72111.png)  

**Code for IBM Examplei**
![image](https://user-images.githubusercontent.com/5246077/162847501-e7eb48e2-46cc-4487-9b22-71cfb2a74a6e.png)

**Comment by dougnlamb** [comment]
I disabled SQL in the settings and now the source files and members are displayed.

Searching source files does **not** work.  Would this be due to disabling SQL? What features do I miss with "Enable SQL" is turned off.

**Reply by worksofliam** [reply]
Sorry, I'm just seeing this. Not too many features, though I am not sure why the search does not work. Perhaps you could say the output from the Code for IBM i Output tab when you try to search.

**Reply by dougnlamb** [reply]
```
.: system "DSPFD FILE(GAL6SH/QSRC) TYPE(*MBR) OUTPUT(*OUTFILE) OUTFILE(ILEDITOR/O_YW1OrAeG)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPF9861: Output file O_YW1ORAEG created in library ILEDITOR.\nCPF9862: Member O_YW1ORAEG added to output file O_YW1ORAEG in library ILEDITOR.\nCPF3030: 753 records added to member O_YW1ORAEG in file O_YW1ORAEG in ILEDITOR.",
    "stderr": ""
}

.: system "QSYS/CPYTOIMPF FROMFILE(ILEDITOR/O_YW1OrAeG O_YW1OrAeG) TOSTMF('/tmp/vscodetemp-O_NeOyrrUJ') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM(',') DECPNT(*PERIOD) "
{
    "code": 0,
    "signal": null,
    "stdout": "CPC2958: All records copied from file O_YW1ORAEG in ILEDITOR.",
    "stderr": ""
}

.: /QOpenSys/usr/bin/qsh
/usr/bin/grep -in -F "Eval" /QSYS.LIB/GAL6SH.LIB/QSRC.FILE/*
{
    "code": 9,
    "signal": null,
    "stdout": "",
    "stderr": ""
}
```

**Reply by worksofliam** [reply]
Got to figure out what status code 9 means. What happens if you try to run the same command from a pase shell? (ssh)

**Reply by dougnlamb** [reply]
No error and no response:
```
$ /usr/bin/grep -in -F "Eval" /QSYS.LIB/GAL6SH.LIB/QSRC.FILE/*
$
```

**Reply by worksofliam** [reply]
Then this is a deeper issue on the IBM i. I'm sure I've seen something like this before. Will have to look into it.

**Reply by dougnlamb** [reply]
Our AS400 admin set the PASE_LANG environment variable to 'EN_US' at the system level.  He also installed db2util.  Now it works.  I believe it was the PASE_LANG variable though.

**Reply by worksofliam** [reply]
Nice! Db2util is no longer used. Defo the language bar!

**Reply by dalea** [reply]
In case the solution does not work for others: 
~~
I'd just started seeing this issue this week.
Reviewed previous notes/recommendations here (notes below) and did not resolve.
Continuing to review, but seeing if anyone can spot my miss...
~
There were no previous issues with the member list or edit of members.  
Our qccsid running with English/37.
Tried the envvar update with PASE_LANG and seeing no difference.  
Tried that selection above and seeing same (no error / no ouput): 
```
call qp2term

/usr/bin/grep -in -F "Eval" /QSYS.LIB/DASHER.LIB/QDDSSRC.FILE/*
$
```
Unchecked extension config for 'Enable SQL' and restarted.
Still unresolved.
Checking with admin for the i to see if we've applied any system updates...
There were some for PASE that had not been applied.
Performed IPL and retesting connections.
Still no joy.
Filter will display objects in the lib, just no list of members for pf-src files...

**Reply by dalea** [reply]
Found solution in #654 (Object Type *SRCPF).
After having no previous issues with my filters (defined with Obj Type *FILE) showing members and allowing open/edit/save, changing the Obj Type for the Filter to *SRCPF allowed the list to resolve.
I have another filter with Type *ALL.  The *file pf-src that are show in the list don't allow expansion to see the members. Something in the filter may not distinguish the type & attributes sufficiently to get/use the member details?

**Comment by dougnlamb** [comment, marked as answer]
It turns out "Enable SQL" is not working because our IBM i is set to 65535 (we're hoping to change this).  So, this is not the cause of the reported problem.

As reported above, the fix to the problem was setting PASE_LANG environment variable to EN_US.
